lib/ui/views/messenger/bloc/discussion_details_form.dart

```diff
@@ -0,0 +1,125 @@
+part of 'providers.dart';
+
+final _discussionDetailsFormProvider = NotifierProvider.autoDispose<
+    DiscussionDetailsFormNotifier, DiscussionDetailsFormState>(
+  () {
+    return DiscussionDetailsFormNotifier();
+  },
+);
+
+@freezed
+class DiscussionDetailsFormState with _$DiscussionDetailsFormState {
+  const factory DiscussionDetailsFormState({
+    @Default('') String name,
+    @Default('') String discussionAddress,
+    @Default([]) List<String> members,
+    @Default([]) List<String> admins,
+  }) = _DiscussionDetailsFormState;
+  const DiscussionDetailsFormState._();
+}
+
+class DiscussionDetailsFormNotifier
+    extends AutoDisposeNotifier<DiscussionDetailsFormState> {
+  DiscussionDetailsFormNotifier();
+
+  late AppLocalizations localizations;
+
+  @override
+  DiscussionDetailsFormState build() => const DiscussionDetailsFormState();
+
+  void init(Discussion discussion) {
+    setName(discussion.name ?? '');
+    setAddress(discussion.address);
+    _initMembers(discussion.membersPubKeys);
+    _initAdmins(discussion.adminsPubKeys);
+  }
+
+  void _initMembers(List<String> members) {
+    for (final member in members) {
```

Comment:
members.forEach(addMember);

Reply:
It's all about calling the addMember or addAdmin sub-functions, that are updating the state of the notifier. So I think that I am going to not change anything.